### DISCUSSION TITLE
DOI Improvements

### DISCUSSION THREAD

**Discussion by gabestein** [knowledgefutures/pubpub#2126] [Ideas]
Hey folks,

We're about to embark on making improvements to our DOI depositing capabilities to better support depositing best practices and patch up a few things on the backend. So far we're planning to include:

- Support for Datacite
- Ask communities to contact us to enable DOI depositing, to combat potential DOI spam
- Depositing abstracts
- Depositing citations
- Depositing license information
- Support more resource types/article types, and make it more configurable
- Option to mint a DOI as part of the release flow
- Option to mint a new version DOI on subsequent releases
- Ability to mint a placeholder DOI prior to release
- Some other behind-the-scenes stuff to improve security and ability to add Crossref/Datacite credentials directly to PubPub

While we're scoping it out, I figured I'd ask if there are other features, metadata fields, etc. you'd like us to consider as part of the project. One thing we're unlikely to do is support fully custom XML, as validating it would be very difficult and we and fear it would lead to increased failed deposits that we'd struggle to support.

**Comment by alicetiara** [comment]
Hi there-- I'm not sure if this has already been addressed, but the ability to have DOIs for individual book chapters or individual conference papers would be A++. Thanks!

**Reply by gabestein** [reply]
Hi! Yup, you can already mint a DOI at the individual pub level in Pub settings.

**Comment by jeffpooley** [comment]
So thrilled for these updates! The new version DOI will be a crucial support for versioning, in the [manner of Zenodo](https://www.openaire.eu/blogs/zenodo-now-supports-doi-versioning-1). Can’t wait for citation and abstract support too. PubPub’s DOI interface is already the best, friendliest to not tech folks I’ve ever encountered.

One question: Will citations contained in footnotes be included? I’m guessing not, at least for those footnotes that use the rich text field. But what if, say, bibtex is used in a footnote’s structured data field?

**Reply by gabestein** [reply]
Thanks for the feedback! I'll be interested to see the uptake of Zenodo-style versioning. Personally, I think it can at times be overused/get expensive, and I know it's more of a Datacite thing than a Crossref thing. But it'll be good to have the option for sure.

Re footnotes: correct, I think they likely won't be included. We have the structured field in footnotes so that you can use bibtex in them, basically, which some communities wanted years ago. But as the name hopefully implies, they're not really meant to be citations in the way e.g. Crossref understands them. Would definitely take suggestions for clarifying that in the UI, or for any ways we could hack it a bit. Crossref/Datacite have no guidance/understanding of footnotes at all that I can find! So, idk, I guess we could tack them on to the end of citations or something if folks thought it was reasonable behavior.

**Reply by jeffpooley** [reply]
Thanks Gabe! I agree about Zenodo-style versioning can get out of hand. 

That makes total sense about footnotes and Crossref DOI inclusion, since footnotes can contain exposition in addition to references/citations. Still, this might raise a bigger issue about footnotes: for many humanities-oriented publications, references appear in footnotes (alongside, sometimes, an end-of-article bibliography listing). Like [Chicago Notes & Bibliography](https://www.chicagomanualofstyle.org/tools_citationguide/citation-guide-1.html). Right now there’s no way that I know of on PubPub for footnote-centric citation approach to take advantage of PubPub’s gorgeous, built-in “Citations” footer. Our workaround has been to just include the bibliography in the main Pub text, as [here](https://doi.org/10.32376/d895a0ea.ed348e03). With the slated DOI improvements, though, there’s no way that I can think of to have the references register in a DOI deposit.

I can think of a couple of ways this ability—to grant access to PubPub’s Citations feature for non-inline/in-text citations styles common in the humanities—could be implemented. One would be to enable the inline system within the footnote window, so that a year, for example (using the “custom” option in the “Inline Format” dropdown menu in the inline window), could be the anchor for the citation. A second option would be to enable, in the Pub Settings, the manual adding of citations, with both rich text and structured data options—so that a rich-text list of references, or a bibtex string, could be pasted in? A third option would be to build this in to the DOI deposit interface, along similar lines (a rich text + structured data field for pasting).

**Reply by gabestein** [reply]
Thanks Jeff. Can you say more about "grant access to PubPub’s Citations feature for non-inline/in-text citations styles common in the humanities?" Would it be sufficient just to have a citation style that uses footnote-like superscript numbering, rather than the current `[1]` style?

I suppose we could also extract any structured cites from footnotes and add them to the references section/deposits. I suspect that could lead to some conflicts from people using both systems, but I suspect that would be rare.

**Reply by jeffpooley** [reply]
So the best-case scenario (which I certainly don’t expect) would be a way to include footnote citations (1) in PubPub’s Citations footer and in the (2) DOI deposits. Under the existing system, there’s great support for putting the footnote-style citations in PubPub’s footnotes, but there’s no way to get them (in their different, [bibliography-style format](https://www.chicagomanualofstyle.org/tools_citationguide/citation-guide-1.html) into the Citations footer (and, by extension, included in a future DOI deposit). Our workaround has been to just include the alphabetized Bibliography—where the references appear a second time, in their different formatting—in the main PubPub text, as [here](https://doi.org/10.32376/d895a0ea.ed348e03).

The idea of a citation style that just used footnote-like superscript numbering is attractive, but one complication is that footnotes include discursive text *and* references. The other option of extracting structured cites from footnotes is also tricky, at least as far as I can tell, since (1) the Structured Data field can only take one reference, it appears (I’ve tested with a multiple-reference bibtex record); and (2) footnotes often include a mix of written text and multiple references, moving back and forth. 

This messiness is the only reason I was floating what is probably unworkable: somehow anchoring a reference to, say, a year in a reference *in a footnote*. This would involve somehow merging the footnote and [inline] citation windows, which sounds … unlikely. So the other option I floated would be to have something like a button in Pub Settings, under Citation Style (renamed Citations), maybe “Advanced,” that would—not unlike the way Customize CSS works now—open up to a page with two fields for a manual citation dump: Rich text (where each line would be parsed by the Citation footer as a citation) and Structured data (for the full list BibTex entries, etc).

It’s crushing my soul even to contemplate that UI clutter, so I’m definitely not expecting any changes to address what may be an edge issue. What I suspect we’ll do is add citations to the Crossref deposit post-deposit.

**Reply by gabestein** [reply]
Ah, thanks! I think I understand now. We’ve had a related problem with a few monographs that have, for lack of a better term, un-cited or more general references in bibliographies or notes. I’ll discuss it with the devs next week and we’ll see what we come up with. All your options seem reasonable, and folks may have other bright ideas.

**Reply by jeffpooley** [reply]
Thanks Gabe!
